### ISSUE TITLE
Investigate hail size plugin

### ISSUE THREAD

**Issue by MoseleyS** [metoppv/improver#1832]
Late in the autumn release, I spotted that the hail size data being generated on the global domain were all zero, with a knock-on dearth of non-zero data in the hail fraction and hail accumulation data. Limited investigation showed that the method for deriving the pressure of the 268K isotherm was returning mostly missing data instead of mostly valid data. This uses the external `stratify` module.

Acceptance criteria:
 * Issue understood and resolved
 * Output hail size for any given model run of global data contain some non-zero data.

**Comment by MoseleyS** [comment]
A bit of digging into data in a notebook has revealed the following insight: Temperature on pressure levels include some missing data points, represented by `nan`. These probably occur everywhere where the near-surface pressure levels are actually below the modelled surface. If I omit these points when calling the `stratify.interpolate` method, I get the result I expect.

Therefore, I need to add handling of missing, masked or `nan` values to the appropriate plugin.

**Comment by MoseleyS** [comment]
I have code that I think works. Now I need an end-to-end suite test to show that large hail are generated. The fact that the acceptance test data have not changed gives me hope.

**Comment by MoseleyS** [comment]
Suite test running in my account with the name `imengl_hail`. (Failures are because I manually killed some tasks when I realised that I hadn't installed the code from my PR!)

The result is hail! This is realization 0 from the 18Z run 25th Jan. The maximum hail stone size is 120 mm. All zero values are coloured white, so the non-zeroes are easier to see.
![image](https://user-images.githubusercontent.com/23079172/214795117-d706d071-196e-4a50-b2a1-39843887234a.png)
